sql/RFC_SQL.md

```diff
@@ -0,0 +1,1165 @@
+console.log(res.rows[0].solution);
+```
+
+**Deno: sqlite** ([link](https://github.com/denodrivers/sqlite3))
+
+```ts
+import { Database } from "jsr:@db/sqlite@0.11";
+
+const db = new Database("test.db");
+
+const [solution] = db.prepare("SELECT 1 + 1 AS solution").value<[string]>()!;
+console.log(solution);
+db.close();
+```
+
+**Deno: postgres** ([link](https://github.com/denodrivers/postgres))
+
+```ts
+import { Client } from "https://deno.land/x/postgres/mod.ts";
+const client = new Client({
+  user: "user",
+  database: "test",
+  hostname: "localhost",
+  port: 5432,
+});
+await client.connect();
+const result = await client.queryObject`SELECT 1 + 1 AS solution`;
+console.log(result.rows[0].solution);
+await client.end();
+```
+
+## Scope
+
+This specification covers the essential components and functionalities required
+for interacting with SQL-based databases through a standardized interface. It
+includes, but is not limited to:
+
+- Connection management
+- Query execution
+- Transaction handling
+- Error handling and reporting
+- Data type mappings
+- Prepared statements and parameterized queries
+
+> Other functionalities such as subscriptions would be out of scope for the
+> first version, but would be considered for upcoming spec releases.
+
+## Goals and Non-Goals
+
+- Define a clear and comprehensive API for database drivers that can be
+  universally applied to all SQL-based databases.
+- Ensure that the interface is flexible enough to support both basic and
+  advanced SQL database functionalities, and that the interfaces can be extended
+  for functionality that is not included in the specs.
+- Promote the adoption of the standardized interface within the developer
+  community and across database vendors.
+- This specification does not aim to replace existing database drivers but
+  rather to provide a layer of standardization that can be implemented by them.
+- It does not cover non-SQL databases or seek to address database-specific
+  optimizations and extensions that fall outside the scope of standard SQL
+  operations. This should be handled by the respective drivers.
+
+## Audience
+
+This RFC is intended for database driver developers, application developers,
+database administrators, and other stakeholders involved in the development and
+maintenance of applications that interact with SQL-based databases. It provides
+a framework for creating compatible and standardized database drivers,
+facilitating smoother development and integration processes.
+
+## Specification
+
+The following section contains the main interfaces introduced in this RFC.
+
+### Concept
+
+#### Initiation of client
+
+There are two types of clients, a Client used for a single connection, and a
+ClientPool for when a pool of clients is needed. Both of these provide the same
+base signature, although the `options` argument differs slightly in signature.
+
+A client class takes two arguments, the first is a `connectionUrl` in the form
```

Comment:
that would be up to the respective drivers. The `connectionUrl` it would be either a string or a URL, and further options could then be added to the url (as JDBC), or be added to the options object depending on the maintainers.

> Why not keep just the connection options?

Because there are differences between every driver as to what they need of options (thus the second argument can be extended), and i find this cleaner and more extendable